tasks/pdp/task_init_pp.go

```diff
@@ -0,0 +1,237 @@
+	ctx := context.Background()
+
+	// Select the proof set where challenge_request_task_id = taskID
+	var proofSet struct {
+		ID int64 `db:"id"`
+	}
+
+	err = ipp.db.QueryRow(ctx, `
+        SELECT id
+        FROM pdp_proof_sets
+        WHERE challenge_request_task_id = $1 
+    `, taskID).Scan(&proofSet)
+	if err == sql.ErrNoRows {
+		// No matching proof set, task is done (something weird happened, and e.g another task was spawned in place of this one)
+		return true, nil
+	}
+	if err != nil {
+		return false, xerrors.Errorf("failed to query pdp_proof_sets: %w", err)
+	}
+
+	// Get the listener address for this proof set from the PDPVerifier contract
+	pdpVerifier, err := contract.NewPDPVerifier(contract.ContractAddresses().PDPVerifier, ipp.ethClient)
+	if err != nil {
+		return false, xerrors.Errorf("failed to instantiate PDPVerifier contract: %w", err)
+	}
+
+	listenerAddr, err := pdpVerifier.GetProofSetListener(nil, big.NewInt(proofSet.ID))
+	if err != nil {
+		return false, xerrors.Errorf("failed to get listener address for proof set %d: %w", proofSet.ID, err)
+	}
+
+	// Determine the next challenge window start by consulting the listener
+	provingSchedule, err := contract.NewIPDPProvingSchedule(listenerAddr, ipp.ethClient)
+	if err != nil {
+		return false, xerrors.Errorf("failed to create proving schedule binding, check that listener has proving schedule methods: %w", err)
+	}
+	init_prove_at, err := provingSchedule.InitChallengeWindowStart(nil, big.NewInt(proofSet.ID))
+	if err != nil {
+		return false, xerrors.Errorf("failed to get next challenge window start: %w", err)
+	}
+
+	// Instantiate the PDPVerifier contract
+	pdpContracts := contract.ContractAddresses()
+	pdpVeriferAddress := pdpContracts.PDPVerifier
+
+	// Prepare the transaction data
+	abiData, err := contract.PDPVerifierMetaData.GetAbi()
+	if err != nil {
+		return false, xerrors.Errorf("failed to get PDPVerifier ABI: %w", err)
+	}
+
+	data, err := abiData.Pack("nextProvingPeriod", big.NewInt(proofSet.ID), init_prove_at, []byte{})
+	if err != nil {
+		return false, xerrors.Errorf("failed to pack data: %w", err)
+	}
+
+	// Prepare the transaction
+	txEth := types.NewTransaction(
+		0,                 // nonce (will be set by sender)
+		pdpVeriferAddress, // to
+		big.NewInt(0),     // value
+		0,                 // gasLimit (to be estimated)
+		nil,               // gasPrice (to be set by sender)
+		data,              // data
+	)
+
+	if !stillOwned() {
+		// Task was abandoned, don't send the transaction
+		return false, nil
+	}
+
+	fromAddress, _, err := pdpVerifier.GetProofSetOwner(nil, big.NewInt(proofSet.ID))
+	if err != nil {
+		return false, xerrors.Errorf("failed to get default sender address: %w", err)
+	}
+
+	// Get the current tipset
+	ts, err := ipp.fil.ChainHead(ctx)
+	if err != nil {
+		return false, xerrors.Errorf("failed to get chain head: %w", err)
+	}
+
+	// Send the transaction
+	reason := "pdp-proving-period"
```

Comment:
Those should be unique for each send in the code. Will be later used to display message type stats to users in the UI.

```suggestion
	reason := "pdp-proving-init"
```